packages/opentelemetry-metrics/src/Metric.ts

```diff
@@ -80,133 +75,28 @@ export abstract class Metric<T extends BaseBoundInstrument>
     this._instruments.clear();
   }
 
-  getMetricRecord(): MetricRecord[] {
-    return Array.from(this._instruments.values()).map(instrument => ({
-      descriptor: this._descriptor,
-      labels: instrument.getLabels(),
-      aggregator: instrument.getAggregator(),
-      resource: this.resource,
-    }));
+  getMetricRecord(): Promise<MetricRecord[]> {
+    return new Promise(resolve => {
+      resolve(
+        Array.from(this._instruments.values()).map(instrument => ({
+          descriptor: this._descriptor,
+          labels: instrument.getLabels(),
+          aggregator: instrument.getAggregator(),
+          resource: this.resource,
+        }))
+      );
+    });
   }
 
   private _getMetricDescriptor(): MetricDescriptor {
     return {
       name: this._name,
-      description: this._options.description,
-      unit: this._options.unit,
+      description: this._options.description || '',
+      unit: this._options.unit || '1',
```

Comment:
It looks like this was default value previously existed here, but what does `'1'` mean for a unit? From the [spec](https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/metrics/api.md#metric-event-format) it seems like just a normal unit of measure, e.g. GBs or whatever. Am I misinterpreting the unit descriptor?

Reply:
this is what our spec says and that's why I added here

https://github.com/open-telemetry/opentelemetry-js/blob/master/packages/opentelemetry-api/src/metrics/Metric.ts#L37
@mayurkale22 can you comment on that ?

Reply:
I think default `'1'` is meant to show that by default it is a single count of something, but you do not know what. Could be 1 second (timer), could be 1 transaction (counter), could be one byte, kb, mb, or gb. Without having a unit provided it is basically impossible to infer what the value should be, so it becomes "1 of _something_".